openjdk/jdk
# JDK-8290011: IGV: Upgrade to Netbeans Platform 14 and remove dead code

Upgrade IGV to the newest Netbeans Platform 14 which was released on June 9 2022 (officially support running on JDK 11 and JDK 17).

## Testing

Tested the following use cases manually on macOS and JDK 17:

- build with maven 3.8.1
- import graphs via network (localhost)
- Save all groups to XML
- Save selected groups to XML
- Remove selected graphs
- Remove selected groups
- Remove all groups
- Open XML graph file
- Expand groups in Outline
- Open a graphs in from same and different group in Outline
- "Open clone" in the Outline
- "Open Difference to current graph" for graphs in same and different group in Outline
- Opening a new graph : Updates the Bytecode and Control Flow window
- Show next / previous graph in current group buttons
- Expand / Reduce the difference selection buttons
- Changing of the difference selection by modifying the slider 
- Extract set of selected nodes and check if they are centered
- Hiding of selected nodes
- Showing all nodes again
- Zooming in / out
- Different views: Sea of nodes /  clustered seas of nodes / CFG
- Satellite view: button and by pressing the S key
- Enable / Disable "Show neighbouring nodes of fully visible nodes semi-transparent"
- Undo / Redo
- Selection mode: button and by holding Ctrl + mouse-drag
- Searching a node: Selects the node and centres it. Makes the node visible if it is hidden
- Searching a block: Selects all nodes in the block and centres it.  Makes the all the nodes in the block visible
- Selecting node(s): adjusts colours in slider. Show property in Properties window
- Hovering a node: highlights node and shows property box
- Hovering a connection: highlights connection and corresponding nodes 
- apply filters
- select nodes corresponding to a bytecode
- select nodes corresponding to a basic block in the control flow

# Minor Fixes
- Fixed that Bytecode Component and Control Flow Component gets updated immediately when opening a new graph
- Fixed warning about missing position attribute of menu items

# Refactoring
- removed lots of unused or redundant code


---------
### Progress

### Issue
 * [JDK-8290011](https://bugs.openjdk.org/browse/JDK-8290011): IGV: Upgrade to Netbeans Platform 14 and remove dead code


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10164/head:pull/10164` \
`$ git checkout pull/10164`

Update a local copy of the PR: \
`$ git checkout pull/10164` \
`$ git pull https://git.openjdk.org/jdk pull/10164/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10164`

View PR using the GUI difftool: \
`$ git pr show -t 10164`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10164.diff">https://git.openjdk.org/jdk/pull/10164.diff</a>

</details>
